tools/deploy-cli/deploy-cli.go

```diff
@@ -0,0 +1,324 @@
+	if err != nil {
+		return "", err
+	}
+
+	if !newlyCreated {
+		log.Printf("[%s] Reading content from existing file: %s", varName, filePath)
+	}
+
+	content, err := ReadFileContent(varName, filePath)
+	if err != nil {
+		return "", err
+	}
+
+	return content, nil
+}
+
+// BuildKustomizeManifest builds a provided kustomize overlays to output, same as `kustomize build`
+func BuildKustomizeManifest(source string) ([]byte, error) {
+	kustomizer := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
+	fSys := filesys.MakeFsOnDisk()
+	resources, err := kustomizer.Run(fSys, source)
+	if err != nil {
+		return nil, err
+	}
+	return resources.AsYaml()
+}
+
+// RenderEmbedTemplateToFile reads in a go-template, renders it with supporting data
+// and then write the result to an output file
+func RenderEmbedTemplateToFile(templateFiles embed.FS, inputFile, outputFile string, data interface{}) error {
+	tmpl, err := template.ParseFS(templateFiles, inputFile)
+	if err != nil {
+		return err
+	}
+	f, err := os.Create(outputFile)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	if err = tmpl.Execute(f, data); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// deployIronic generates the YAML for Ironic using Kustomize and applies it to the Kubernetes cluster.
+func deployIronic(data *DeployContext, tempIronicOverlay string) error {
+	if data.IronicHostIP == "" {
+		return fmt.Errorf("failed to determine IRONIC_HOST_IP")
+	}
+	ironicKustomizeTpl := "templates/ironic-kustomize.tpl"
+	kustomizeFile := filepath.Join(tempIronicOverlay, "kustomization.yaml")
+
+	if err := RenderEmbedTemplateToFile(data.TemplateFiles, ironicKustomizeTpl, kustomizeFile, data); err != nil {
+		return err
+	}
+
+	ironicBMOConfigMapTpl := "templates/ironic_bmo_configmap_env.tpl"
+	ironicBMOConfigMapOutput := filepath.Join(tempIronicOverlay, "ironic_bmo_configmap.env")
+
+	if err := RenderEmbedTemplateToFile(data.TemplateFiles, ironicBMOConfigMapTpl, ironicBMOConfigMapOutput, data); err != nil {
+		return err
+	}
+
+	return deployWithKustomizeAndApply(tempIronicOverlay)
+}
+
+// deployBMO generates the YAML for the Bare Metal Operator using Kustomize
+// and applies it to the Kubernetes cluster.
+func deployBMO(data *DeployContext, tempBMOOverlay string) error {
+
+	inputFile := "templates/bmo-kustomize.tpl"
+	kustomizeFile := filepath.Join(tempBMOOverlay, "kustomization.yaml")
+	if err := RenderEmbedTemplateToFile(data.TemplateFiles, inputFile, kustomizeFile, data); err != nil {
+		return err
+	}
+	ironicEnvSrc := "https://raw.githubusercontent.com/metal3-io/baremetal-operator/main/config/default/ironic.env"
+	ironicEnvDst := filepath.Join(tempBMOOverlay, "ironic.env")
+	if err := DownloadFile(ironicEnvSrc, ironicEnvDst); err != nil {
+		return err
+	}
+
+	return deployWithKustomizeAndApply(tempBMOOverlay)
+}
+
+// getKubeconfigPath determines the kubeconfig path from KUBECTL_ARGS, KUBECONFIG_PATH, or defaults to ~/.kube/config.
+func getKubeconfigPath() (string, error) {
```

Comment:
I think we have the same in e2e. Let's reuse the code or make it reusable if necessary!